### PR TITLE
Add r-ragtop

### PR DESCRIPTION
**Pull request by TanguyGen**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by TanguyGen** [comment]
@conda-forge/help-r

**Comment by bgruening** [comment]
Please check your PR. You have some commits included about r-admm

**Comment by TanguyGen** [comment]
This branch was made from the r-admm branch  but I supressed all files for r-admm to replace them by those from ragtop so there  are no files remaining about r-admm

**Comment by mfansler** [comment]
@TanguyGen merging in `main` fixes this. In the future, it's always best practice to synchronize your `main` branch against upstream (`conda-forge/staged-recipes`) and start a new branch from that.

If you're frequently contributing, you might be interested in the scripts that I use: https://gist.github.com/mfansler/b7b75f73495b940948efa3c1f58827bf.

**Comment by TanguyGen** [comment]
Nice thanks ! It was my last one before a while but I will save it for later

File: recipes/r-ragtop/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    - r-limsolve >=1.5.5.1
+
+test:
+  commands:
+    - $R -e "library('ragtop')"           # [not win]
+    - "\"%R%\" -e \"library('ragtop')\""  # [win]
+
+about:
+  home: https://CRAN.R-project.org/package=ragtop
+  license: GPL-2.0-only
```
Comment (mfansler) [inline recipes/r-ragtop/meta.yaml:43]:
```suggestion
  license: GPL-2.0-or-later
```

File: recipes/r-ragtop/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  summary: Algorithms to price American and European equity options, convertible bonds and a
+    variety of other financial derivatives. It uses an extension of the usual Black-Scholes
+    model in which jump to default may occur at a probability specified by a power-law
+    link between stock price and hazard rate as found in the paper by Takahashi, Kobayashi,
+    and Nakagawa (2001) <doi:10.3905/jfi.2001.319302>.  We use ideas and techniques
+    from Andersen and Buffum (2002) <doi:10.2139/ssrn.355308> and Linetsky (2006) <doi:10.1111/j.1467-9965.2006.00271.x>.
+  license_family: GPL2
+  license_file:
+    - '{{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2'
```
Comment (mfansler) [inline recipes/r-ragtop/meta.yaml:52]:
```suggestion
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-2
    - {{ environ["PREFIX"] }}/lib/R/share/licenses/GPL-3
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for contributing! Please adjust license, then should be good.

**Review by bgruening — APPROVED** [review]

**Review by mfansler — APPROVED** [review]

**Pull request merged**